R/network_analysis.R

```diff
@@ -51,24 +52,28 @@ make_res <- function(dat, period_select = NULL,
      stop("region_select not in regions of data")
       }
    }
-
-  if(is.null(sector_select) == F & "sector" %in% names(dat) == F){
-    stop("Data missing sector information. Define sectors")
-  }
-
- if(is.null(region_select)){
+ #if region not specified, select all regions
+  if(is.null(region_select)){
    region_select <- unique(dat %>%
                              dplyr::select(region) %>%
                              tidyr::drop_na()) %>%
      dplyr::pull(region)
  }
+#if there is no sector information, append a dummy sector
+ if("sector" %in% names(dat) == F){
+   dat <- dat %>%
+     mutate(sector = "null_sector")
+ }
+ #if sector not specified, select all sectors
  if(is.null(sector_select)){
    sector_select <- unique(dat %>%
```

Comment:
If you would like to keep everything in the tidyverse style, can I recommend %>% dplyr::distinct() %>% after the drop_na() call instead of wrapping the output in a unique() function?